### ISSUE TITLE
Option to select network interface!

### ISSUE THREAD

**Issue by Nonie689** [igo95862/bubblejail#43]
### Description

Can you insert an option to select an network interface to be used?
And an option to set the used IP to be used in that interface?

**Comment by igo95862** [comment]
It is probably possible with [slirp4netns](https://github.com/rootless-containers/slirp4netns) but seems to be pretty difficult because of how bwrap creates namespaces.

**Comment by Nonie689** [comment]
Maybe would it be a workaround to create an own bubblejail user or group...

I found this topic,

https://askubuntu.com/questions/1204412/changing-the-network-interface-for-specific-user

could this work?

**Comment by igo95862** [comment]
> Maybe would it be a workaround to create an own bubblejail user or group...
> 
> I found this topic,
> 
> https://askubuntu.com/questions/1204412/changing-the-network-interface-for-specific-user
> 
> could this work?

No. Modifying iptables would require root access

**Comment by igo95862** [comment]
> FYI: https://codeberg.org/crabjail/crabjail/src/commit/6d4e5076c74e7e29afe377b296ff723291d9344f/src/crabjail.rs#L698
> 
> I did not looked at the architecture of bubblejail in depth yet but a simple slirp4netns integration is rather simple I guess.
> 
> ```
> term1$ bwrap --dev-bind / / --unshare-net bash
> term1-sandboxed$ echo $$ > /tmp/pid
> term2$ slirp4netns --configure $(cat /tmp/pid) tap0
> term1-sandboxed$ wget -qO- 1.1.1.1
> ... long html ...
> ```
> 
>     * To get the pid/namespace from outside and when `--unshare-pid` is used, `--json-status-fd` can be used.
> 
>     * If `/etc/resolv.conf` points to `127.0.0.1`, DNS will not work.
> 
>     * As printed by slirp4netns: `WARNING: 127.0.0.1:* on the host is accessible as 10.0.2.2 (set --disable-host-loopback to prohibit connecting to 127.0.0.1:*)`
> 
>     * Hardening option are not used `--enable-sandbox`, `--enable-seccomp`, ...
> 
>     * Deeper configuration is complex.

Yes, however it will no longer work if you use `--dev` option of bwrap.

```
term1$ bwrap --dev /dev_test --dev-bind / / --unshare-net bash
term2$ slirp4netns --configure $(cat /tmp/pid) tap0
setns(CLONE_NEWNET): Operation not permitted
child failed(1)
```

I believe it is caused by bwrap using nested namespaces when `--dev` is used.

https://github.com/util-linux/util-linux/issues/1592

**Comment by rusty-snake** [comment]
Actually you will use the PID reported by `--json-status-fd`. Using a process inside the sandbox is a lot extra work and `--unshare-pid` destroy everything (and `--dev` too as it looks like.

So let's try the example with `--json-status-fd`:

```console
term1$ bwrap --dev-bind / / --unshare-net --json-status-fd 3 bash 3<&1        
{ "child-pid": 29620, "mnt-namespace": 4026534194, "net-namespace": 4026534195 }
term2$ slirp4netns --configure 29620 tap0
term1-sandboxed$ wget -qO- 1.1.1.1
... long html ...
```

**Comment by igo95862** [comment]
I think I found a solution for bubblewrap namespace issues. If you attach at least one process to the parent user names space it becomes targetable though pid. `slirp4netns --userns-path /proc/[target_pid]/ns/user --configure [sandboxed_pid] tap0` 

See the util-linux issue I linked.

However, there is another problem. With `/etc/resolv.conf`. slirp4netns puts DNS over the `10.0.2.3` address but the `resolv.conf` is going to be the host one. Probably a unionfs can be used to overlay custom files for `/etc` but it is not simple to implement since bubblewrap does not support unionfs out of the box.
